src/coreclr/vm/jithelpers.cpp

```diff
@@ -5714,7 +5714,7 @@ HCIMPL3(void, JIT_VTableProfile32, Object* obj, CORINFO_METHOD_HANDLE baseMethod
     WORD slot = pBaseMD->GetSlot();
     _ASSERTE(slot < pBaseMD->GetMethodTable()->GetNumVirtuals());
 
-    MethodDesc* pMD = pMT->GetMethodDescForSlot(slot);
+    MethodDesc* pMD = pMT->GetMethodDescForSlot_NoThrow(slot);
```

Comment:
Does this mean it can return `NULL`? If so, what is protecting that below?

Reply:
No, it returns what it would have before, but its a slightly different algorithm that doesn't use any allocation. Its a bit slower, but perf results indicates that it is ok.